ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -135,10 +135,12 @@ HierarchyMathOps::HierarchyMathOps(std::string name,
       d_coarsest_ln(coarsest_ln),
       d_finest_ln(finest_ln),
       d_fc_var(new FaceVariable<NDIM, double>(d_object_name + "::scratch_fc")),
-      d_nc_var(new NodeVariable<NDIM, double>(d_object_name + "::scratch_nc", NDIM)),
+      d_nc_s_var(new NodeVariable<NDIM, double>(d_object_name + "::scratch_nc_s", 1)),
+      d_nc_v_var(new NodeVariable<NDIM, double>(d_object_name + "::scratch_nc_v", NDIM)),
       d_sc_var(new SideVariable<NDIM, double>(d_object_name + "::scratch_sc")),
       d_of_var(new OuterfaceVariable<NDIM, double>(d_object_name + "::scratch_of")),
-      d_on_var(new OuternodeVariable<NDIM, double>(d_object_name + "::scratch_on", NDIM == 2 ? 1 : NDIM)),
+      d_on_s_var(new OuternodeVariable<NDIM, double>(d_object_name + "::scratch_on_s", 1)),
+      d_on_v_var(new OuternodeVariable<NDIM, double>(d_object_name + "::scratch_on_v", NDIM)),
```

Comment:
Its ugly, but this should make everything work for both scalars and vectors in 2D and 3D.